dune/gdt/operators/advection-fv-entropybased.hh

```diff
@@ -23,63 +23,128 @@ namespace GDT {
 
 template <class OperatorImp, class InverseHessianOperatorImp>
 class EntropicCoordinatesOperator
-  : public OperatorInterface<typename OperatorImp::MatrixType, typename OperatorImp::SGV, OperatorImp::s_r>
+  : public OperatorInterface<typename OperatorImp::AGV,
+                             OperatorImp::s_r,
+                             OperatorImp::s_rC,
+                             OperatorImp::r_r,
+                             OperatorImp::r_rC,
+                             typename OperatorImp::F,
+                             typename OperatorImp::M,
+                             typename OperatorImp::SGV,
+                             typename OperatorImp::RGV>
 {
-  using BaseType = OperatorInterface<typename OperatorImp::MatrixType, typename OperatorImp::SGV, OperatorImp::s_r>;
-
 public:
+  using ThisType = EntropicCoordinatesOperator;
+  using BaseType = OperatorInterface<typename OperatorImp::AGV,
+                                     OperatorImp::s_r,
+                                     OperatorImp::s_rC,
+                                     OperatorImp::r_r,
+                                     OperatorImp::r_rC,
+                                     typename OperatorImp::F,
+                                     typename OperatorImp::M,
+                                     typename OperatorImp::SGV,
+                                     typename OperatorImp::RGV>;
+
   using typename BaseType::RangeSpaceType;
+  using typename BaseType::SourceFunctionType;
   using typename BaseType::SourceSpaceType;
   using typename BaseType::VectorType;
 
   using OperatorType = OperatorImp;
   using InverseHessianOperatorType = InverseHessianOperatorImp;
 
-  EntropicCoordinatesOperator(const OperatorType& operator_in,
-                              const InverseHessianOperatorType& inverse_hessian_operator)
-    : operator_(operator_in)
+  EntropicCoordinatesOperator(const OperatorType& opertr, const InverseHessianOperatorType& inverse_hessian_operator)
+    : BaseType(opertr.parameter_type() + inverse_hessian_operator.parameter_type())
+    , operator_(opertr)
     , inverse_hessian_operator_(inverse_hessian_operator)
   {}
 
+  // pull in methods from various base classes
+  using BaseType::apply;
+
+  /// \name Required by ForwardOperatorInterface.
+  /// \{
+
+  const SourceSpaceType& source_space() const override final
+  {
+    return operator_.source_space();
+  }
+
   bool linear() const override final
   {
     return false;
   }
 
-  const SourceSpaceType& source_space() const override final
+  // avoid non-optimal default implementation in OperatorInterface
+  void apply(SourceFunctionType source_function,
+             VectorType& range_vector,
+             const XT::Common::Parameter& param = {}) const override final
   {
-    return operator_.source_space();
+    this->assert_matching_range(range_vector);
+    VectorType u_update = range_vector.copy();
+    u_update.set_all(0.);
```

Comment:
Could this be changed to init with 0 and correct size? Also in a few other places I think.

Reply:
Yes, agreed.